### DISCUSSION TITLE
nelder mead optimizer unable to find resonable solution

### DISCUSSION THREAD

**Discussion by dmrowan** [phoebe-project/phoebe2#517] [Questions]
Hello, 

I am using the nelder mead optimizer to fit model parameters to a detached eclipsing binary. This shows the best solution I have been able to obtain for data in the SDSS:g passband: 
![dmrowan_github_discussion](https://user-images.githubusercontent.com/36493412/130986588-d1072412-461e-4799-9ef0-3ceffd0309ec.png)

This fit does not accurately describe the depth and width of the secondary eclipse near orbital phase -0.2. The shape of the out-of-eclipse variability near phase 0 is also incorrect. 

Here I used the ellc backend with reflection turned on for the maximum number of iterations and function evaluations. I found a similar solution when using the phoebe backend. The free parameters are the teffs, requivfracs, ecc, per0, and incl@binary. I have also experimented with using the teffratio and requivsumfrac as free parameters and found a similar solution. 

I think a better solution can be obtained by adjusting the temperatures and radii, but have been unable to find such a solution with the nelder mead optimizer. I have this same issue for similar light curves where the primary eclipse is much deeper than than the secondary. Are there any steps I can take to push the optimizer in the right direction for these cases?

I can also provide the bundle for this example if that would be helpful.

**Comment by aprsa** [comment]
Can you tell us more about the system? The width of the secondary seems to be significantly larger than that of the primary. While usually that is a telltale of eccentricity, I'd imagine in this case it might be something else, just guessing from the shape of the lightcurve.

**Reply by dmrowan** [reply]
Sure, these are the fit parameters after optimization: 
      teff@primary@star@component: 5069.798880339088 K
      teff@secondary@star@component: 3500.0000779489665 K
      requivfrac@primary@star@com...: 0.1294814972089363 solRad
      requivfrac@secondary@star@c...: 0.22968905616315521 solRad
      ecc@binary@orbit@component: 0.0036070307761843484
      per0@binary@orbit@component: 79.47560408813985 deg
      incl@binary@orbit@component: 83.61120329045632 deg

The optimization itself terminates after ~1200 iterations with success=True. 

Here is a VizieR link to this system: http://vizier.u-strasbg.fr/viz-bin/VizieR-4?-c=205.57114-43.09282&-c.rs=5&-out.add=_r&-sort=_r 

I think the secondary teff is hitting some lower limit due to the limb darkening tables (I am using ld_mode='lookup'). I can kick and fix the primary teff to something higher, say 10000K, and the fit ends up looking the same. Gaia DR2 estimates the teff at ~7200. 

I have also tried using the semidetached constraint on the secondary, thinking that maybe making the secondary as large as possible would save some trouble if the optimizer was stuck near the requiv_max limit. This actually made the fit _worse_ -- it misses the secondary entirely and makes the out-of-eclipse variability essentially flat.

**Reply by aprsa** [reply]
In general, I would not fit both temperatures simultaneously unless you have calibrated multi-color lightcurves. Instead, I'd fit teff2/teff1 as an independent parameter and stick with that.

I'm trying to figure out what kind of an object this is; it looks semi-detached by just looking at the lightcurve, which imposes e=0. If that's the case, then what you might be seeing here is an accretion disk effect, which ellc (nor phoebe) doesn't support. Then no matter what you do, phoebe will never give you a good fit simply because other parameters are trying to compensate for unsupported physics. If I had to blindly guess what's happening here, that would be my guess.

What is the orbital period of this binary?

**Reply by dmrowan** [reply]
The orbital period is 3.514 days. 

I have tried to model similar eclipsing binaries where the primary is much deeper than the secondary, but I am having similar problems. This makes me think that even if an accretion disk is partially responsible for this example, there is still something I am doing incorrectly with the optimization. 

AO Ser has a light curve that is quite similar to this example (see Figure 3 of [Alton & Prsa, 2012](http://underoakobservatory.com/FreshLookAtThe%20Algol-like%20Eclipsing%20Binary%20AO%20Ser.pdf))

It seems like data from my example is most similar to the B band data of AO Ser, while the phoebe fit model is similar to the I band data of AO Ser. Could that mean that something is incorrect with temperatures, reflection or limb darkening coefficients in my model?

**Comment by aprsa** [comment]
AO Ser (as best as I remember ;-)) had equal width eclipses. Depth change was significant across colors but the durations were unaffected. At the 3.5-day period I might retract my accretion disk idea -- it seems too long for a semi-detached configuration.

As far as the Nelder-Mead minimizer is concerned, it'll contract onto a local minimum bound by the initial simplex. If the actual minimum is not encompassed, then NMS won't stand a chance of getting you there. Exploring the parameter space by hand might be a better first step, and once you're close, you're likely to benefit from sampling the topology of the parameter space with MCMC. (for the record, that's abusing mcmc to do something it's not designed to do, but I actually think it's okay in this case because it will inform you of the topology and thus the range of parameters worth exploring).

**Comment by aprsa** [comment]
Whoops, I forgot to mention that, of all parameters that might help, I'd put my bet on the mass ratio. Try to sample over it and see whether it helps with the overall lightcurve shape.

**Reply by dmrowan** [reply]
Ah, that seems to have done it, thanks! I did another round of optimization allowing just the masses and teff ratio to vary, after adopting the solution from above, and things look much better:

![55010538_success](https://user-images.githubusercontent.com/36493412/131035811-40ff9ea7-d646-4fdd-9f99-a1515d5788ff.png)

The edges of the secondary aren't quite right, but certainly an improvement, especially with the out-of-eclipse shape. 

The masses came out to 
      mass@primary@star@component: 1.122576698494004 solMass
     mass@secondary@star@component: 0.24135107697727398 solMass
